src/data/priv_appendable_data.rs

```diff
@@ -384,296 +323,256 @@ mod test {
+    //                                   Some(&keys1.1)));
+    //     assert_eq!(orig_priv_appendable_data.add_signature(&keys2.1).ok(),
+    //                Some(0));
+
+    //     // Update with wrong version
+    //     let mut wrong_version = unwrap!(PrivAppendableData::new(name,
+    //                                                             2,
+    //                                                             vec![new_owner.0],
+    //                                                             owner_keys.clone(),
+    //                                                             BTreeSet::new(),
+    //                                                             Filter::white_list(None),
+    //                                                             encrypt_keys.0,
+    //                                                             Some(&keys1.1)));
+    //     assert_eq!(wrong_version.add_signature(&keys2.1).ok(), Some(0));
+    //     // Shall throw error of UnknownMessageType
+    //     assert!(orig_priv_appendable_data.update_with_other(wrong_version).is_err());
+
+    //     // Update with owner_keys in different order
+    //     let mut wrong_order = unwrap!(PrivAppendableData::new(name,
+    //                                                           1,
+    //                                                           vec![new_owner.0],
+    //                                                           vec![keys3.0, keys2.0, keys1.0],
+    //                                                           BTreeSet::new(),
+    //                                                           Filter::white_list(None),
+    //                                                           encrypt_keys.0,
+    //                                                           Some(&keys1.1)));
+    //     assert_eq!(wrong_order.add_signature(&keys2.1).ok(), Some(0));
+    //     // Shall throw error of UnknownMessageType
+    //     assert!(orig_priv_appendable_data.update_with_other(wrong_order).is_err());
+
+    //     // Update with wrong identifier
+    //     let mut wrong_name = unwrap!(PrivAppendableData::new(rand::random(),
+    //                                                          1,
+    //                                                          vec![new_owner.0],
+    //                                                          owner_keys,
+    //                                                          BTreeSet::new(),
+    //                                                          Filter::white_list(None),
+    //                                                          encrypt_keys.0,
+    //                                                          Some(&keys1.1)));
+    //     assert_eq!(wrong_name.add_signature(&keys2.1).ok(), Some(0));
+    //     // Shall throw error of UnknownMessageType
+    //     assert!(orig_priv_appendable_data.update_with_other(wrong_name).is_err());
+    // }
```

Comment:
Not sure about that. It will depend on how we deduplicate the logic and whether our set of data types changes, etc. And we can always retrieve them from git history.

But if you feel that it will be less work to update them than to rewrite them, I'd prefer to update them immediately, whenever we do the changes, and keep them `#[ignore]`d instead of commented.